File: src/main/java/reflection/dymanicloading/commands/Fib.java
package reflection.dymanicloading.commands;

import java.util.List;
import java.util.function.Function;

public class Fib implements Function<List<String>, String> {

    @Override public String apply(List<String> strings) {
        int num = Integer.parseInt(strings.getFirst());

        int result = fib(num);

        return String.valueOf(result);
    }

    int fib(int num) {
        if (num < 2) return 1;
        return fib(num - 1) + fib(num - 2);
    }
}

File: src/main/java/interpreter/fib.js
/**
 * fib.js
 *
 * @param {array} args
 * @return {string}
 */

const fib = (n) => {
    if(n < 2) return 1
    else return fib(n-1) + fib(n-2)
}
const result = fib(args[0])

String(result)
